posthog/tasks/warehouse.py

```diff
@@ -114,3 +114,28 @@ def capture_workspace_rows_synced_by_team(team_id: int) -> None:
     team.save()
 
     ph_client.shutdown()
+
+
+@shared_task(ignore_result=True)
+def validate_data_warehouse_table_columns(team_id: int, table_id: str) -> None:
+    from posthog.warehouse.models import DataWarehouseTable
+
+    ph_client = get_ph_client()
+
+    try:
+        table = DataWarehouseTable.objects.get(team_id=team_id, id=table_id)
+        for column in table.columns.keys():
+            table.columns[column]["valid"] = table.validate_column_type(column)
+        table.save()
+
+        ph_client.capture(team_id, "validate_data_warehouse_table_columns succeeded")
+    except Exception as e:
+        logger.exception(
```

Comment:
What happens on the UI if it fails?

Reply:
No difference from now - the invalid field becomes non-queryable (so that you can still query the other fields), and you can update the schema from the schema edit modal